docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,882 @@
+
+    ```yaml
+    apiVersion: frrk8s.metallb.io/v1beta1
+    kind: FRRConfiguration
+    metadata:
+      name: blue
+      namespace: frr-k8s-system
+      # A custom label
+      labels:
+        network: blue
+    spec:
+      bgp:
+        routers:
+        - asn: 64512
+          neighbors:
+          - address: 182.18.0.5
+            asn: 64512
+            disableMP: true
+            holdTime: 1m30s
+            keepaliveTime: 30s
+            passwordSecret: {}
+            port: 179
+            toAdvertise:
+              allowed:
+                mode: filtered
+            toReceive:
+              # Allow to accept the routes to remote nodeSubnets.
+              allowed:
+                mode: filtered
+                prefixes:
+                - ge: 24
+                  le: 24
+                  prefix: 10.10.0.0/16
+    ```
+
+4. The cluster admin advertises the CUDN pod network. In this example, the
+   `targetVRF` is set to default, meaning routes to the podNetworks will be
+   advertised to the default VRF.
+
+    ```yaml
+    apiVersion: k8s.ovn.org/v1
+    kind: RouteAdvertisements
+    metadata:
+      name: blue
+    spec:
+      # advertise routes to the default VRF
+      targetVRF: default
+      # nodeSelector must be empty, since we don't support a network in a overlay and no-overlay hybrid mode.
+      nodeSelector: {}
+      frrConfigurationSelector:
+        matchLabels:
+        # Select the FRRConfiguration defined in step-3
+        network: blue
+      networkSelectors:
+      - networkSelectionType: ClusterUserDefinedNetwork
+        clusterUserDefinedNetworkSelector:
+          networkSelector:
+            matchLabels:
+              # Select the CUDN defined in step-2
+              network: blue
+      advertisements:
+      - PodNetwork
+    ```
+
+5. OVN-Kubernetes will generate the following FRRConfiguration for each node.
+
+    ```yaml
+    apiVersion: frrk8s.metallb.io/v1beta1
+    kind: FRRConfiguration
+    metadata:
+      name: route-generated-blue
+      namespace: frr-k8s-system
+    spec:
+      bgp:
+        routers:
+        - asn: 64512
+          neighbors:
+          - address: 182.18.0.5
+            disableMP: true
+            asn: 64512
+            toAdvertise:
+              allowed:
+                prefixes:
+                # the local node subnet
+                - 10.10.1.0/24
+            toReceive:
+              # Allow to receive the routes to remote nodeSubnets.
+              # This is no-overlay specific behavior.
+              allowed:
+                mode: filtered
+                prefixes:
+                - ge: 24
+                  le: 24
+                  prefix: 10.10.0.0/16
+          prefixes:
+          - 10.10.1.0/24
+    ```
+
+### Deployment Consideration
+
+#### BGP Topology
+
+The selected `FRRConfiguration` CR determines the deployment mode for the
+no-overlay network. There will be no new field added to the FRRConfiguration.
```

Comment:
I think we are currently using RA API as the link to make changes - for me that's an extra user step.. shouldn't we do the configs ourself internally if the no-overlay-mode flag is set to none for that network?